gateway/ln-gateway/src/gatewayd/lnrpc_client.rs

```diff
@@ -0,0 +1,130 @@
+use std::{fmt::Debug, sync::Arc};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fedimint_api::dyn_newtype_define;
+use fedimint_server::modules::ln::route_hints::RouteHint;
+use tonic::{
+    transport::{Channel, Endpoint},
+    Request, Streaming,
+};
+use tracing::error;
+use url::Url;
+
+use crate::{
+    gatewaylnrpc::{
+        gateway_lightning_client::GatewayLightningClient, CompleteHtlcsRequest,
+        CompleteHtlcsResponse, GetPubKeyRequest, GetPubKeyResponse, PayInvoiceRequest,
+        PayInvoiceResponse, SubscribeInterceptHtlcsRequest, SubscribeInterceptHtlcsResponse,
+    },
+    LnGatewayError, Result,
+};
+
+// TODO: Issue 1554: Define gatewaylnpc spec for getting route hints
+pub struct GetRouteHintsResponse {
+    pub route_hints: Vec<RouteHint>,
+}
+
+#[async_trait]
+pub trait ILnRpcClient: Debug + Send + Sync {
+    /// Get the public key of the lightning node
+    async fn get_pubkey(&self) -> Result<GetPubKeyResponse>;
```

Comment:
nit: just naming suggestion. but feel free to move it to a later point I guess.
- `pubkey()`
- `route_hints()`
- `pay(invoice)`  (we had that already :'D)

Reply:
adopted. Plus, renamed `subscribe_intercept_htlcs` to `subscribe_htlcs`. simple